src/HomeWork/StudentGrade.java

```diff
@@ -0,0 +1,41 @@
+package HomeWork;
+
+import java.util.Scanner;
+import java.util.TreeMap;
+
+public class StudentGrade {
+    public static void main(String[] args) {
+        boolean isExit;
+        do {
+            try {
+                TreeMap<Integer, String> gradeMap = new TreeMap<>();
```

Comment:
Please try to create a separate class MarkDefiner and move all logic related to the defining mark (validation is included) there